Fix test failures induced by setting the SOURCE_DATE_EPOCH variable.

Taken from upstream: <https://github.com/p11-glue/p11-kit/pull/213>.

From 2a474e1fe8f4bd8b4ed7622e5cf3b2718a202562 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <dueno@redhat.com>
Date: Mon, 28 Jan 2019 13:03:15 +0100
Subject: [PATCH] extract-jks: Prefer _p11_extract_jks_timestamp to
 SOURCE_DATE_EPOCH

Give _p11_extract_jks_timestamp precedence over SOURCE_DATE_EPOCH so
that the test results are not affected by the envvar settings.
---
 trust/extract-jks.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/trust/extract-jks.c b/trust/extract-jks.c
index ad8dc35..a6f855f 100644
--- a/trust/extract-jks.c
+++ b/trust/extract-jks.c
@@ -250,7 +250,9 @@ prepare_jks_buffer (p11_enumerate *ex,
 	 * when this was this certificate was added to the keystore, however
 	 * we don't have that information. Java uses time in milliseconds
 	 */
-	{
+	if (_p11_extract_jks_timestamp)
+		now = _p11_extract_jks_timestamp;
+	else {
 		char *source_date_epoch;
 		source_date_epoch = secure_getenv ("SOURCE_DATE_EPOCH");
 		if (source_date_epoch) {
@@ -276,9 +278,7 @@ prepare_jks_buffer (p11_enumerate *ex,
 				return false;
 			}
 			now = epoch;
-		} else if (_p11_extract_jks_timestamp)
-			now = _p11_extract_jks_timestamp;
-		else
+		} else
 			now = time (NULL);
 	}
 
